react/components/ListsPage/index.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { Component, Fragment, ReactNode } from 'react'
+import { withRuntimeContext } from 'vtex.render-runtime'
+import { Spinner } from 'vtex.styleguide'
+
+import ApolloClient from 'apollo-client'
+import { concat, filter, findIndex, map, update } from 'ramda'
+import { withApollo, WithApolloClient } from 'react-apollo'
+import { InjectedIntlProps, injectIntl, IntlShape } from 'react-intl'
+
+import {
+  createList,
+  getListsFromLocaleStorage,
+  saveListIdInLocalStorage,
+} from '../../GraphqlClient'
+
+import Content from './Content'
+import ListSelector from './ListSelector'
+
+import wishlist from '../../wishList.css'
+
+const ON_LISTS_PAGE_CLASS = 'vtex-lists-page'
+
+interface ListsPageState {
+  lists?: any
+  selectedListId?: string
+  isLoading?: boolean
+}
+
+interface ListsPageProps {
+  client: ApolloClient<any>
+  runtime: any
+  intl: IntlShape
+}
+
+class ListsPage extends Component<ListsPageProps & WithApolloClient<any> & InjectedIntlProps, ListsPageState> {
+  public state: ListsPageState = {
+    isLoading: true,
+  }
+  private isComponentMounted: boolean = false
+
+  public componentWillUnmount(): void {
+    this.isComponentMounted = false
+    document.body.classList.remove(ON_LISTS_PAGE_CLASS)
+  }
+
+  public componentDidUpdate(prevProps: any): void {
+    if (this.props !== prevProps) {
```

Comment:
you should only check for the props you are actually *using*. this check is same the as not having it, because the props *object* will change on every update.